frontend/src/features/admin-form/settings/SettingsPage.tsx

```diff
@@ -50,6 +52,25 @@ export const SettingsPage = (): JSX.Element => {
   const displayPayments =
     user?.betaFlags?.payment || flags?.has(featureFlags.payment)
 
+  // Note: Admins are not redirected to /general on invalid settings tabs.
+  useEffect(() => {
+    if (displayPayments) {
```

Comment:
Hmm I feel like the way we are handling `payments` in `settingsTabOrder` is a bit convoluted. Looking ahead, once we bring payments into actual prod, this will need a bigger refactor too

Why not lets just keep it within `settingsTabOrder` and then check `displayPayments` when accessing `settingsTab === 'payments'`

Reply:
Yea valid point, it did cross my mind to do this instead but I was thinking how this will reveal our beta features HAHA But okay I think that was just paranoia speaking and I don't think admins will try to enumerate our pages 😂

Reply:
Hm though, we might pre-maturely redirect them to general because the displayPayments will take time :/ okay let me see what I can do

Reply:
okay actually another concern I had was also in the case where we have multiple tabs where we may hide from users, cos in that case, just using the array itself won't work cos its a fixed order (and won't correspond to the tab index).

Reply:
Honestly, I think it isn't that bad to remove this conditional when we move to payments on prod? Just need to remove the `useEffect` hook

Reply:
Yup on second thought I think its okay to keep the hooks around, don't think we can circumvent it as we want to delay the redirect under `displayPayments` is properly loaded. 

Though I still think we should keep payments in the original `settingsTabOrder`, And just do a conditional check within the useEffect!

Also do you think we should also just check and redirect erroneous settingsTab (like `settings/whatever` to `settings/general`)? I don't think it will usually happen though so might not be a cause for concern

Reply:
Discussed to keep current implementation but add comments for documenting rationale

Reply:
As discussed over huddle, we'll keep the current implementation for extensibility. Shouldn't matter since payments to prod should happen soon!! 🤞 🤞

Reply:
comments added in [1c38c7f](https://github.com/opengovsg/FormSG/pull/6348/commits/1c38c7f887dee8ceb4fa2547f67544a8afd77fd1)